lib/kennel/models/record.rb

```diff
@@ -142,18 +136,10 @@ def build_json
 
       def build
         @unfiltered_validation_errors = []
-        json = nil
-
-        begin
-          json = build_json
-          (id = json.delete(:id)) && json[:id] = id
-          validate_json(json)
-        rescue StandardError
-          if unfiltered_validation_errors.empty?
-            @unfiltered_validation_errors = nil
-            raise PrepareError, safe_tracking_id # FIXME: this makes errors hard to debug when running tests
-          end
-        end
+
```

Comment:
I'm not convinced that this is a good idea.

In case it wasn't clear, the purpose of this code is as follows: validation errors _should_ nicely raise an exception via `invalid!`; the build / validation code shouldn't raise other exceptions. This is what allows us to report _all_ of the problems to the user, rather than trickle-feeding one problem at a time.

However: the purpose of this code is this: if a record has already raised an error via `invalid!`, then we "forgive" any subsequent crashes from the validation code for this item, since the item is obviously already known to be in an untrusted state.

Reply:
my thinking was:
there should not be exceptions in the validation code because that means the library made  mistake -> fix the library instead of hiding the problem (and making it harder to fix)

Reply:
... and if exceptions are expected then any exception should be captured and not just "exceptions after validation errors" so it's consistent